docs/pages/image-extractor.rst

```diff
@@ -145,23 +147,94 @@ and DataLoader, refer to this guide: https://pytorch.org/tutorials/beginner/data
 
     for i, sample_batch in enumerate(dataloader):
         img, label = sample_batch["rgba"], sample_batch["label"]
-        img = img.permute(0, 3, 1, 2)  # Reshape to PyTorch format for convolutions
+        img = img.permute(0, 3, 1, 2)
         out = net(img)
 
 
+`Writing Custom Pose Extractors`_
+=================================
+
+The pose extractor defines how camera poses are programmatically determined so that the image
+extractor knows how to manipulated the camera position and angle to extract an image from the
+simulator. All custom pose extractors must inherit from the PoseExtractor abstract class and
+override the extract_poses method. Further, you must register the pose extractor using
+habitat_sim.registry.
+
+The extract_poses method takes three required arguments:
+
+* labels: List[float]: A list of numeric labels of points we are looking for
+* view: numpy.ndarray: A 2 dimensional array representing the topdown view of a scene.
+* fp: str: The filepath of the scene.
+
+The job of the extract_poses method is to return a list of poses where each pose is a four-tuple
+of (<camera position>, <point of interest>, <label>, <scene filepath>). The camera position is the
+coordinates of the camera in the space of the topdown view. The point of interest is the position where
+the camera will be pointing in the space of the topdown view. The label and filepath are the class
+label of what the camera is pointing at and the scene filepath the pose comes from respectively. Below
+is an example of a pose extractor that simply chooses some random navigable points and looks forward.
+
+.. code:: py
+    import numpy as np
+    import habitat_sim.registry as registry
+
+    from habitat_sim.utils.data import ImageExtractor, PoseExtractor
+
+    @registry.register_pose_extractor(name="random_pose_extractor")
+    class RandomPoseExtractor(PoseExtractor):
+        def __init__(self, topdown_views, pixels_per_meter=0.1):
+            super().__init__(topdown_views, pixels_per_meter)
+
+        def extract_poses(self, labels, view, fp):
+            height, width = view.shape
+            num_random_points = 4
+            points = []
+            while len(points) < num_random_points:
+                row, col = np.random.randint(0, height), np.random.randint(0, width)
+
+                # Convenient method to check if a point is navigable
+                if self._valid_point(row, col, view):
+                    points.append((row, col))
+
+            poses = []
+            for point in points:
+                r, c = point
+                point_of_interest = (r - 1, c) # Just look forward
+                pose = (point, point_of_interest, 0.0, fp) # Different labels not currently supported, so just pass a 0.0 through
```

Comment:
Is the label of interest coming? If not, we should state this above.

Reply:
The label of interest is dependent on being able to generate a topdown map of semantic labels. I opened up an issue for this a while back, but I don't think the is as simple as generating the topdown map of navigable points. But anyway, you're right I think its best to remove this for now.